packages/http/src/Adaptor.js

```diff
@@ -43,87 +35,69 @@ export function execute(...operations) {
   };
 }
 
-var cookiejar = new tough.CookieJar();
-var Cookie = tough.Cookie;
+function handleResponse(state, response) {
+  return {
+    ...composeNextState(state, response.body),
+    response,
+  };
+}
 
-axios.interceptors.request.use(config => {
-  cookiejar?.getCookies(config.url, (err, cookies) => {
-    config.headers.cookie = cookies?.join('; ');
-  });
-  return config;
-});
+function handleCallback(state, callback) {
```

Comment:
Here's a tip  - if you default the callback argument, this gets a lot easier
```
function handleCallback(state, callback = (s) => s) {
    return callback(state)
}
```
I actually think modern javascript has a few good approaches around this sort of pattern, including `return callback?.(state) ?? state`